components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/applications/ApplicationUtils.java

```diff
@@ -191,52 +128,56 @@ public static StringBuilder getTextPayload (String appId, String groupName, Stri
                                             String hostName, String repoUrl, String alias, Map<String, String> customPayloadEntries, String[] dependencyAliases, 
                                             org.apache.stratos.common.Properties properties, String oauthToken,String[] dependencyClusterIDs,
                                             String[] exportMetadata, String[] importMetadata)
             throws ApplicationDefinitionException {
 
         //Create the payload
-        BasicPayloadData basicPayloadData = createBasicPayload(appId, groupName, cartridgeInfo, subscriptionKey,
+        BasicPayloadData basicPayloadData = createBasicPayload(appId, groupName, cartridge, subscriptionKey,
                 clusterId, hostName, repoUrl, alias, tenantId, dependencyAliases, dependencyClusterIDs,exportMetadata,importMetadata);
         //Populate the basic payload details
         basicPayloadData.populatePayload();
 
-        PayloadData payloadData = PayloadFactory.getPayloadDataInstance(cartridgeInfo.getProvider(),
-                cartridgeInfo.getType(), basicPayloadData);
-
-        // get the payload parameters defined in the cartridgeInfo definition file for this cartridgeInfo type
-        if (cartridgeInfo.getProperties() != null && cartridgeInfo.getProperties().length != 0) {
-
-            org.apache.stratos.common.Properties cartridgeProps = AutoscalerUtil.toCommonProperties(cartridgeInfo.getProperties());
-            
-            if (cartridgeProps != null) {
-
-                for (Property propertyEntry : cartridgeProps.getProperties()) {
-                    // check if a property is related to the payload. Currently
-                    // this is done by checking if the
-                    // property name starts with 'payload_parameter.' suffix. If
-                    // so the payload param name will
-                    // be taken as the substring from the index of '.' to the
-                    // end of the property name.
-                    if (propertyEntry.getName().startsWith("payload_parameter.")) {
-                        String payloadParamName = propertyEntry.getName();
-                        String payloadParamSubstring = payloadParamName.substring(payloadParamName.indexOf(".") + 1);
-                        if ("DEPLOYMENT".equals(payloadParamSubstring)) {
-                            payloadData.getBasicPayloadData().setDeployment(payloadParamSubstring);
-                            continue;
+        PayloadData payloadData = PayloadFactory.getPayloadDataInstance(cartridge.getProvider(),
+                cartridge.getType(), basicPayloadData);
+
+        // get the payload parameters defined in the cartridge definition file for this cartridge type
+
+        if (cartridge.getProperties() != null) {
+            if (cartridge.getProperties().getProperties() != null && cartridge.getProperties().getProperties().length != 0) {
```

Comment:
Maybe we'll have to refactor the code properly to make the method call "cartridge.getProperties().getProperties()" to look clean. Using more context sensitive names upon the corresponding methods would do, I guess.